scripts/landcover.py

```diff
@@ -0,0 +1,202 @@
+                        props[str(classname)] = int(numpy.sum(pixels == key))
+                    props["name"] = tile.properties.get("Name")
+                    props["count"] = int(len(numpy.unique(pixels)))
+
+                    result.append(
+                        {
+                            "geometry": dict(tile["geometry"]),
+                            "properties": props,
+                        }
+                    )
+
+    with fiona.open(
+        Path(wd, "mgrs_stats.fgb"),
+        "w",
+        driver="FlatGeobuf",
+        crs=WGS84,
+        schema=SCHEMA,
+    ) as colxn:
+        colxn.writerecords(result)
+
+
+def split_highest(data, column, size, pool=1000, seed=RANDOM_SEED):
+    """
+    Split hightest values of a cloumn from a dataframe.
+    """
+    data.sort_values(column, ascending=False, inplace=True)
+    return data[:pool].sample(size, random_state=seed)
+
+
+def percentages(data):
+    """
+    Normalize all numerical columns to percentages
+    """
+    data_num = data.select_dtypes(include="number")
+    data_norm = data_num.div(data_num.sum(axis=1), axis=0)
+    data[data_norm.columns] = data_norm
+
+    return data
+
+
+def sample(wd):
+    """
+    Sample the mgrs tiles based on landcover statistics.
+
+    Target: ~1000 tiles
+    Set very small counts to zero. Exclude high latitudes.
+    200 samples from the 2000 most diverse
+    50 samples from the 1000 highest for all other categories except water
+    100 samples from all tiles with water between 30% an 70% (making sure we
+    capture some, but exclude only purely water so we catch coasts)
+    """
+    data = geopandas.read_file(Path(wd, "mgrs_stats.fgb"))
+
+    data_norm = percentages(data.loc[:, data.columns != "count"])
+    data[data_norm.columns] = data_norm
+
+    diversity = split_highest(data, "count", 200, 2000)
+    urban = split_highest(data, "Built-up", 200)
+    wetland = split_highest(data, "Herbaceous wetland", 50)
+    mangroves = split_highest(data, "Mangroves", 50)
+    moss = split_highest(data, "Moss and lichen", 50)
+    cropland = split_highest(data, "Cropland", 50)
+    trees = split_highest(data, "Tree cover", 50)
+    shrubland = split_highest(data, "Shrubland", 50)
+    grassland = split_highest(data, "Grassland", 50)
+    bare = split_highest(data, "Bare / sparse vegetation", 50)
+    snow = split_highest(data, "Snow and Ice", 50)
+
+    selector = numpy.logical_and(
+        data["Permanent water bodies"] > WATER_LOWER_TH,
+        data["Permanent water bodies"] < WATER_UPPER_TH,
+    )
+    water = data[selector].sample(100, random_state=RANDOM_SEED)
+
+    result = pandas.concat(
+        [
+            diversity,
+            urban,
+            wetland,
+            mangroves,
+            moss,
+            cropland,
+            trees,
+            shrubland,
+            grassland,
+            bare,
+            snow,
+            water,
+        ]
+    )
```

Comment:
Just to understand, this sampling function is independently getting the highest values for each category (plus some extra MGRS tiles for diversity and water areas), and then concatenating those rows together into a single dataframe?

Plotting your `mgrs_sample.geojson` file from https://github.com/Clay-foundation/model/pull/29#issuecomment-1812636118, I see a few cases where the exact MGRS tile is sampled more than once. E.g.:

MGRS tile 56VLM - sampled 3 times:

![image](https://github.com/Clay-foundation/model/assets/23487320/db4a2e78-c727-4a04-baa0-c65e96f187b6)

MGRS tile 17RMQ - sampled 2 times:

![image](https://github.com/Clay-foundation/model/assets/23487320/2cb690b6-5145-4cb4-b3f7-ad4c62790387)

MGRS tile 32TLP - sampled 2 times:

![image](https://github.com/Clay-foundation/model/assets/23487320/e54c32f5-1fd3-43a5-936a-7d5158cf2d62)

The duplicates might be due to the independent random sampling per-category and then concatenation. Perhaps we could remove such duplicate rows before saving out the GeoJSON file?

Reply:
Good catch, will drop duplicates as part of the script.